pkg/logs/decoder/decoder.go

```diff
@@ -67,21 +66,29 @@ func NewMessage(content []byte, status string, rawDataLen int, timestamp string)
 	}
 }
 
-// Decoder splits raw data into lines and passes them to a lineParser that passes them to
-// a lineHandler that emits outputs
-// Input->[decoder]->[parser]->[handler]->Message
+// Decoder wraps a collection of internal actors, joined by channels, representing the
+// whole as a single actor with InputChan of type *decoder.Input and OutputChan of type
+// *decoder.Message.
+//
+// Internally, it has three running actors:
```

Comment:
I know this is how the code was structured before, and maybe this isn't the time to change it, but I wonder if we gain anything by running all this in separate goroutines? The pipeline is strictly sequential, we use unbuffered channels, so I wonder if plumbing this with just function calls would lead to simpler code.

Reply:
That's certainly possible, but for the moment I'm focused on refactoring and not changing the architecture.  I think this is the sort of refactoring that makes these questions possible to ask, though!